SofaKernel/modules/SofaBaseMechanics/SubsetMapping.inl

```diff
@@ -169,9 +170,12 @@ void SubsetMapping<TIn, TOut>::init()
 
     topology = this->getContext()->getMeshTopology();
 
-    // Initialize functions and parameters
-    f_indices.createTopologicalEngine(topology);
-    f_indices.registerTopologicalData();
+    if (f_handleTopologyChange.getValue())
+    {
+        // Initialize functions and parameters for topological changes
+        f_indices.createTopologicalEngine(topology);
+        f_indices.registerTopologicalData();
+    }
```

Comment:
This is intended, the developer has the ability to set both flags on or off based on the scene design.
In most of our scenes it's not the responsability of the MechanicalMapping to resize the output model but rather it's the propagation of the topological changes up to each MechanicalState that is in charge of resizing it. The mapping should only handle its own internal data, except if someone else is already doing it (like if this list of indices is linked to the output of an engine such as BoxROI or from an associated TopologicalMapping).